src/ResourceManagement/Sql/SqlSyncGroupImpl.cs

```diff
@@ -0,0 +1,426 @@
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses DNS alias operations.</param>
+        ///GENMHASH:1B84E5430DBD759F48B80D4C7F8781CC:63DD15F38F429D8D42EAECC53ABE710A
+        internal SqlSyncGroupImpl(string name, SqlDatabaseImpl parent, SyncGroupInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, parent)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = parent.ResourceGroupName();
+            this.sqlServerName = parent.SqlServerName();
+            this.sqlDatabaseName = parent.Name();
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name.</param>
+        /// <param name="sqlServerName">The parent SQL server name.</param>
+        /// <param name="sqlDatabaseName">The parent SQL Database name.</param>
+        /// <param name="name">The name of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses DNS alias operations.</param>
+        ///GENMHASH:93B2CDB9BD4EC90C1E3E6788282EF80D:46ED0A0EFDCFB4A8C02A54C6267EEAB9
+        internal SqlSyncGroupImpl(string resourceGroupName, string sqlServerName, string sqlDatabaseName, string name, SyncGroupInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            this.resourceGroupName = resourceGroupName;
+            this.sqlServerName = sqlServerName;
+            this.sqlDatabaseName = sqlDatabaseName;
+        }
+
+        /// <summary>
+        /// Creates an instance of external child resource in-memory.
+        /// </summary>
+        /// <param name="name">The name of this external child resource.</param>
+        /// <param name="innerObject">Reference to the inner object representing this external child resource.</param>
+        /// <param name="sqlServerManager">Reference to the SQL server manager that accesses DNS alias operations.</param>
+        ///GENMHASH:2593A87C1076A6D668BDFE33052F341C:E7C65DE6BF45CF0BA29C2A32A669F3CF
+        internal SqlSyncGroupImpl(string name, SyncGroupInner innerObject, ISqlManager sqlServerManager)
+            : base(innerObject, null)
+        {
+            this.name = name;
+            this.sqlServerManager = sqlServerManager;
+            if (innerObject != null && innerObject.Id != null)
+            {
+                if (innerObject.Id != null)
```

Comment:
looks like redundant check (line 91 already has it)

Reply:
fixed